### PR TITLE
fix/3588

### PR DESCRIPTION
**Pull request by jcnelson**
In `handle_headers()`, properly handle the case where the headers list is empty.

### CODE REVIEW

**Comment by fpbgg** [comment]
The integration test failures look unrelated to this fix.

**Review by fpbgg — APPROVED** [review]

**Review by kantai — COMMENTED** [review]
This needs a unit test and a changelog entry.

**Comment by jcnelson** [comment]
Done

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to the versioning scheme outlined in the [README.md](README.md).
 
+## [Unreleased]
+
+### Fixed
+
+- Handle the case where a bitcoin node returns zero headers (#3588)
```
Comment (kantai) [inline CHANGELOG.md:12]: This won't be `[Unreleased]` once it's merged to master -- it should be `[2.1.0.0.1]`

**Review by kantai — APPROVED** [review]

**Review by obycode — APPROVED** [review]

**Pull request merged**